app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEActivity.java

```diff
@@ -506,13 +508,16 @@ private void observeDownload() {
     }
 
     private void observeLegacyInteractions() {
+        SingleEventEnforcer singleEventEnforcer = new SingleEventEnforcerImpl();
+
         viewModel.getLegacyInteraction().observe(this, legacyInteraction -> {
             if (legacyInteraction != null) {
                 switch (legacyInteraction.getId()) {
-                    case ON_ENROLL_CLICK -> {
+                    case ON_ENROLL_CLICK -> singleEventEnforcer.processEvent( () -> {
```

Comment:
What do you think about move this action to the presenter -> onEnrollClick(...)?
What is the purpose of _LegacyInteraction.OnEnrollClick interaction = (LegacyInteraction.OnEnrollClick) legacyInteraction;_

Reply:
if you look at the code it is casting the event to the type of action on the when. it is present in all the options of Legacy interaction, if you can think of a more legible way of putting this I'm happy to update it. About moving the action to the presenter I think it makes sense for some of them but not for all, I will make my case in each of the comments